IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m

```diff
@@ -114,14 +119,31 @@ + (nullable MSIDWPJKeyPairWithCert *)findWPJRegistrationInfoWithAdditionalPrivat
     queryPrivateKey[(__bridge id)kSecClass] = (__bridge id)kSecClassKey;
     queryPrivateKey[(__bridge id)kSecReturnAttributes] = @YES;
     queryPrivateKey[(__bridge id)kSecReturnRef] = @YES;
+#if !MSID_ENABLE_ECC_SUPPORT
     // TODO: hardcoding this to query RSA keys only for now. Once ECC registration is ready and tested, after removing this line, code should be able to find either ECC or RSA keys, since there should be single key corresponding to the tag per tenant
     queryPrivateKey[(__bridge id)kSecAttrKeyType] = (__bridge id)kSecAttrKeyTypeRSA;
-    
+#endif
     status = SecItemCopyMatching((__bridge CFDictionaryRef)queryPrivateKey, (CFTypeRef*)&privateKeyCFDict); // +1 privateKeyCFDict
     if (status != errSecSuccess)
     {
-        MSID_LOG_WITH_CTX(MSIDLogLevelError, nil, @"Failed to find workplace join private key with status %ld", (long)status);
-        return nil;
+        if (shouldCheckEnclave && privateKeyCFDict == NULL)
+        {
+            // Checking if key exists in Secure Enclave
+            NSData *tagData = queryPrivateKey[(__bridge id)kSecAttrApplicationTag];
+            NSString *tag = [[NSString alloc] initWithData:tagData encoding:NSUTF8StringEncoding];
+            if (![tag hasSuffix:kECPrivateKeyTagSuffix])
```

Comment:
Should this be a task of the caller to specify this? This function could previously query any applicationTag, but new addition assumes it's querying a very specific tag. Any way to improve this?

Reply:
I didn't quite understand your question, can you clarify what you mean by it could query any application tag? Isn't the new tag format <existing wpj private key tag>#<tenant id>-EC for ECC device key? Since the tag is already passed in by caller via queryAttributes, we are just adding -EC at the end so that if key is in secure enclave it will be returned. The code would remain unchanged if both RSA & ECC keys had the same tag.

Reply:
I mean, I could technically query something that has a completely different applicationTag using this function. For example, I could query "com.microsoft.prtprivatekey" tag using this code as well, since it didn't make any assumptions for what is being queried. Now, the code was changed to query specific WPJ private key, since it's checking for a specific WPJ-related EC-tag.

Reply:
This could become a problem if we decide to query another set of keys (e.g. token binding keys) in the future, and keys could be following a different tagging/naming model.

Reply:
It could be more extensible/clean if the caller passed a list of applicationTags to check for.

Reply:
My initial thinking was to keep this function unchanged and have the caller call it twice, once with application tag as for RSA and if that fails, call again with -EC tag, so the onus is on the caller to keep trying tags. But I saw this function being used just for querying DK so added that logic inside. I'll revert the changes to have the caller keep querying with different tag.